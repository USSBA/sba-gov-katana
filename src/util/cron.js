var CronJob = require("cron").CronJob; // eslint-disable-line id-match
import config from "config";

import { followupEmailJob, sendDataToOcaJob } from "../service/linc-service.js";

function init() {
  /* eslint-disable no-new, id-match */
  if (config.get("linc.resendEmailJobEnabled")) {
    console.log("Starting resendConfirmationEmail job");
    new CronJob(config.get("linc.resendEmailCron"), followupEmailJob, null, true, "America/New_York");
  }
  if (config.get("linc.sendToOcaJobEnabled")) {
    console.log("Starting sending failed or pending messages to OCA");
    new CronJob(config.get("linc.ocaSoapCron"), sendDataToOcaJob, null, true, "America/New_York");
  }
/* eslint-enable no-new, id-match */
}

export { init };
