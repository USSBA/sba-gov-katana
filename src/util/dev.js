import HttpStatus from "http-status-codes";

export function enableWebpackHotModuleReplacement(app, silent) {
  console.log("Enabling Webpack");
  const webpack = require("webpack"); // eslint-disable-line global-require
  const webpackConfig = require("../../webpack.config.development")(); // eslint-disable-line global-require
  const compiler = webpack(webpackConfig);
  app.use(
    require("webpack-dev-middleware")(compiler, {
      // eslint-disable-line global-require
      publicPath: webpackConfig.output.publicPath,
      noInfo: silent
    })
  );

  app.use(require("webpack-hot-middleware")(compiler)); // eslint-disable-line global-require
}

export function addDevelopmentErrorHandler(app) {
  app.use(function(err, req, res, next) {
    // eslint-disable-line no-unused-vars
    if (err) {
      console.log(err);
      if (req.xhr) {
        res.status(HttpStatus.INTERNAL_SERVER_ERROR).send({
          error: "Something went wrong! Oh no!"
        });
      } else {
        res.status(err.status || HttpStatus.INTERNAL_SERVER_ERROR);
        res.render("error", {
          message: err.message,
          error: err
        });
      }
    }
  });
}
