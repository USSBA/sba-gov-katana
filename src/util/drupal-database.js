import { fetchDescription, fetchVisibility } from "../models/dao/disaster.js";
import Promise from "bluebird";

function fetchDisasterFromDrupalDatabase() {
  return Promise.all([fetchDescription(), fetchVisibility()])
    .spread((description, visibility) => {
      return {
        description: description,
        visible: visible
      };
    });
}
export { fetchDisasterFromDrupalDatabase };
