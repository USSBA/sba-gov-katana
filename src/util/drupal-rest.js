import axios from "axios";
import _ from "lodash";
import config from "config";
import Promise from "bluebird";
import URL from "url";

var requestObject = {
  baseURL: config.get("drupal.hostname"),
  headers: {
    "Accepts": "application/json"
  }
};


function fetchFromDrupal(url, query) {
  const options = _.merge({}, requestObject);
  options.url = url;
  if (query) {
    options.params = query;
  }
  return axios.request(options)
    .then(function(response) {
      return response.data;
    });
}


function fetchBlogsFromDrupal() {
  return fetchFromDrupal(config.get("drupal.endpoint") + "/recent-blogs")
    .then((blogs) => {
      if (_.isArray(blogs)) {
        return _.map(blogs, (blog, index) => {
          /* eslint-disable no-magic-numbers */
          return {
            url: blog.url,
            title: blog.title,
            name: blog.name,
            date: blog.date,
            imageUrl: index === 0 ? "assets/images/homepage/two-ladies.jpg" : "assets/images/homepage/open-for-business-blogs-photo.jpg"
          };
          /* eslint-enable no-magic-numbers */
        });
      }
      return [];
    });
}


function fetchFrontPageSlidesFromDrupal() {
  return fetchFromDrupal("entityqueue_subqueue.json", {
      name: "frontpage_slide_order"
    })
    .then((result) => {
      return _.map(result.list[0].eq_node, "id");
    })
    .then((ids) => {
      return Promise.map(ids, (nodeId) => {
        return fetchFromDrupal("/node/" + nodeId + ".json");
      });
    })
    .then((nodes) => {
      return _.map(nodes, function(item) {
        const originalUrl = item.field_url.url;
        const fixedUrl = URL.parse(originalUrl).path;
        return {
          fileId: item.field_image.file.id,
          imageAlt: item.field_image.alt,
          url: fixedUrl,
          title: item.title
        };
      });
    })
    .then((slides) => {
      return Promise.map(slides, (slide) => {
        return fetchFromDrupal("/file/" + slide.fileId + ".json")
          .then(function(result) {
            return {
              imageAlt: slide.imageAlt,
              url: slide.url,
              title: slide.title,
              image: result.url
            };
          });
      });
    });
}


export {
  fetchFromDrupal,
  fetchFrontPageSlidesFromDrupal,
  fetchBlogsFromDrupal
};
