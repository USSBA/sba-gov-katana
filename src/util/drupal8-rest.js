import axios from "axios";
import _ from "lodash";
import config from "config";
import Promise from "bluebird";
import URL from "url";

var requestObject = {
  baseURL: config.get("drupal8.hostname"),
  headers: {
    "Accepts": "application/json"
  }
};


function fetchFromDrupal8(type, id) {
  const options = _.merge({}, requestObject);
  options.url = type + "/" + id + "/";
  options.params = {
    "_format": "json"
  };
  console.log(options);
  return axios.request(options)
    .then(function(response) {
      return response.data;
    });
}


function fetchNodeFromDrupal8(nodeId) {
  return fetchFromDrupal8("node", nodeId);
}

function fetchParagraphFromDrupal8(paragraphId) {
  return fetchFromDrupal8("entity/paragraph", paragraphId);
}

function formatDrupal8NodeResponse(response) {
  let paragraphs = response['field_paragraphs'];
  let title = response['title'][0].value;
  return Promise.map(paragraphs, (paragraph) => {
    return fetchParagraphFromDrupal8(paragraph['target_id']);
  }).then((paragraphData) => {
    return {
      title: title,
      paragraphs: paragraphData
    };
  });


}


function fetchFormattedNode(nodeId) {
  return fetchNodeFromDrupal8(nodeId).then(formatDrupal8NodeResponse);
}






export {
  fetchFormattedNode
};
