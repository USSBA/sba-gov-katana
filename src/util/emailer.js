import * as nodemailer from "nodemailer";
import config from "config";
import Promise from "bluebird";
import _ from "lodash";

const sendMailConnectionOptions = {
  sendmail: true,
  newline: "unix",
  path: "/usr/sbin/sendmail"
};

var sesConnectionOptions = {
  secure: true,
  host: config.get("email.hostname"),
  auth: {
    user: config.get("email.username"),
    pass: config.get("email.password")
  },
  logger: true
};

var transporter = nodemailer.createTransport(sesConnectionOptions);

function sendConfirmationEmail(options) {
  return new Promise((resolve, reject) => {
    var defaultMailOptions = {
      from: config.get("email.sender"),
      to: config.get("email.sender"),
      subject: "SBA Test Email",
      text: "Test Email",
      html: "<p>This is a test</p>"
    };
    var mailOptions = _.assign({}, defaultMailOptions, options);
    if (config.get("email.debugEmailOnly")) {
      console.log(
        // eslint-disable-next-line no-magic-numbers
        "Email sender would have sent:" + JSON.stringify(mailOptions, 0, 4)
      );
      resolve();
    } else {
      // send mail with defined transport object
      transporter.sendMail(mailOptions, function(error, info) {
        if (error) {
          reject(error);
        } else {
          resolve(info);
        }
      });
    }
  });
}
export { sendConfirmationEmail };
