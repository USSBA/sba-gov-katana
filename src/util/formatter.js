import sanitizeHtml from "sanitize-html";

function sanitizeTextSectionHtml(dirty) {
  let clean = "";
  try {
    clean = sanitizeHtml(dirty, {
      allowedTags: ["p", "strong", "em", "u", "hr", "a", "ol", "ul", "li", "dl", "dt", "dd", "table", "tbody", "thead", "tfoot", "tr", "td", "th"],
      allowedAttributes: {
        "a": ["href"]
      }
    });
  } catch (error) {
    console.error(error);
  }
  return clean;
}


export { sanitizeTextSectionHtml };
