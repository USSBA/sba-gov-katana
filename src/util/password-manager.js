/* eslint-disable */
import config from "config";
var encryptor = require("simple-encryptor")(config.get("oca.soap.passwordDecryptionKey"));
var generatePassword = require("password-generator");


var maxLength = 12;
var minLength = 8;
var uppercaseMinCount = 1;
var lowercaseMinCount = 1;
var numberMinCount = 1;
var specialMinCount = 1;
var UPPERCASE_RE = /([A-Z])/g;
var LOWERCASE_RE = /([a-z])/g;
var NUMBER_RE = /([\d])/g;
var SPECIAL_CHAR_RE = /([!@#$%^&])/g;
var NON_REPEATING_CHAR_RE = /([\w\d\?\-])\1{2,}/g;

function isStrongEnough(password) {
  var uc = password.match(UPPERCASE_RE);
  var lc = password.match(LOWERCASE_RE);
  var n = password.match(NUMBER_RE);
  var sc = password.match(SPECIAL_CHAR_RE);
  var nr = password.match(NON_REPEATING_CHAR_RE);
  return password.length >= minLength &&
    !nr &&
    uc && uc.length >= uppercaseMinCount &&
    lc && lc.length >= lowercaseMinCount &&
    n && n.length >= numberMinCount &&
    sc && sc.length >= specialMinCount;
}

function generateOcaPassword() {
  var password = "";
  var randomLength = Math.floor(Math.random() * (maxLength - minLength)) + minLength;
  while (!isStrongEnough(password)) {
    password = generatePassword(randomLength, false, /[\w\d!@#$%^&]/);
  }
  return password;
}


function decryptPassword(password) {
  return encryptor.decrypt(password);
}

function encryptPassword(password) {
  return encryptor.encrypt(password);
}

export { generateOcaPassword, decryptPassword, encryptPassword };
