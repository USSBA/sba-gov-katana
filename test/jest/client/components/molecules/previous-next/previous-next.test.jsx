/*global expect*/

/* eslint-disable no-unused-vars,no-undef */
import React from "react";
/*eslint-enable no-unused-vars*/
import PreviousNext from "client/components/molecules/previous-next/previous-next.jsx";
import {shallow} from "enzyme";
import renderer from "react-test-renderer";
import _ from "lodash";
import util from "util";

import lineageBusinessGuide from "../../test-data/lineage-business-guide.json";
import lineageForPartners from "../../test-data/lineage-for-partners.json";

describe("PreviousNext", () => {
  test("Won't render without a lineage", () => {
    // Silence console errors for this test with deliberate errors
    const origErr = console.error
    console.error = jest.fn();
    expect(renderer.create(<PreviousNext />)).toThrowErrorMatchingSnapshot();
    console.error = origErr;
  });
  describe("Business Guide", () => {
    describe("When in a middle article", () => {
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={lineageBusinessGuide}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it sets the desktop previous url appropriately", () => {
        const expectedUrl = "/business-guide/launch/pick-your-business-location-zoning-laws";
        const component = shallow(<PreviousNext lineage={lineageBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-previous-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it sets the desktop next url appropriately", () => {
        const expectedUrl = "/business-guide/launch/choose-your-business-name-register";
        const component = shallow(<PreviousNext lineage={lineageBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-next-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it sets the mobile next url appropriately", () => {
        const expectedUrl = "/business-guide/launch/choose-your-business-name-register";
        const component = shallow(<PreviousNext lineage={lineageBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-next-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
    });
    describe("When in the first article of a middle section", () => {
      const modifiedBusinessGuide = _.merge(_.cloneDeep(lineageBusinessGuide), [{},{},lineageBusinessGuide[1].children[0]]);
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={modifiedBusinessGuide}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it assigns Previous Article to the last article of the previous section", () => {
        const expectedUrl = "/business-guide/plan/buy-existing-business-franchise";
        const component = shallow(<PreviousNext lineage={modifiedBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-previous-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
    });
    describe("When in the last article of a middle section", () => {
      const modifiedBusinessGuide = _.merge(_.cloneDeep(lineageBusinessGuide), [{},{},_.last(lineageBusinessGuide[1].children)]);
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={modifiedBusinessGuide}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it assigns Next Article to the first article of the next section", () => {
        const expectedUrl = "/business-guide/manage/manage-your-finances-business-credit";
        const component = shallow(<PreviousNext lineage={modifiedBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-next-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
    });
    describe("When in the first article of the first section", () => {
      const modifiedBusinessGuide = _.merge(_.cloneDeep(lineageBusinessGuide), [
        {},
        _.first(lineageBusinessGuide[0].children),
        _.first(_.first(lineageBusinessGuide[0].children).children)
      ]);
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={modifiedBusinessGuide}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it sets the desktop next url appropriately", () => {
        const expectedUrl = "/business-guide/plan/write-your-business-plan-template";
        const component = shallow(<PreviousNext lineage={modifiedBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-next-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it sets the mobile next url appropriately", () => {
        const expectedUrl = "/business-guide/plan/write-your-business-plan-template";
        const component = shallow(<PreviousNext lineage={modifiedBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-next-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it does not have a desktop Previous Article element", () => {
        const component = shallow(<PreviousNext lineage={modifiedBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-previous-url").exists()).toBe(false);
      });
    });
    describe("When in the last article of the last section", () => {
      const modifiedBusinessGuide = _.merge(_.cloneDeep(lineageBusinessGuide), [
        {},
        _.last(lineageBusinessGuide[0].children),
        _.last(_.last(lineageBusinessGuide[0].children).children)
      ]);
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={modifiedBusinessGuide}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it sets the desktop previous url appropriately", () => {
        const expectedUrl = "/business-guide/grow/native-american-owned-businesses-programs";
        const component = shallow(<PreviousNext lineage={modifiedBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-previous-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it does not have a Next Article element", () => {
        const component = shallow(<PreviousNext lineage={modifiedBusinessGuide} />);
        expect(component.find("#desktopDivId .previousnext-next-url").exists()).toBe(false);
      });
    });
  });
  describe("For Partners", () => {
    describe("When in a middle article", () => {
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={lineageForPartners}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it sets the desktop previous url appropriately", () => {
        const expectedUrl = "/for-partners/lenders/7a-loans/operate-as-a-7a-lender";
        const component = shallow(<PreviousNext lineage={lineageForPartners} />);
        expect(component.find("#desktopDivId .previousnext-previous-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it sets the desktop next url appropriately", () => {
        const expectedUrl = "/for-partners/lenders/7a-loans/training-and-courses";
        const component = shallow(<PreviousNext lineage={lineageForPartners} />);
        expect(component.find("#desktopDivId .previousnext-next-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it sets the mobile next url appropriately", () => {
        const expectedUrl = "/for-partners/lenders/7a-loans/training-and-courses";
        const component = shallow(<PreviousNext lineage={lineageForPartners} />);
        expect(component.find("#mobileDivId .previousnext-next-url").props().href).toBe(expectedUrl);
      });
    });
    describe("When in the first article of a section", () => {
      const modifiedLineageForPartners = _.merge(_.cloneDeep(lineageForPartners), [{},{},{},lineageForPartners[2].children[0]]);
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={modifiedLineageForPartners}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it does not have a desktop Previous Article element", () => {
        const component = shallow(<PreviousNext lineage={modifiedLineageForPartners} />);
        expect(component.find("#desktopDivId .previousnext-previous-url").exists()).toBe(false);
      });
      test("it sets the desktop next url appropriately", () => {
        const expectedUrl = "/for-partners/lenders/7a-loans/operate-as-a-7a-lender";
        const component = shallow(<PreviousNext lineage={modifiedLineageForPartners} />);
        expect(component.find("#desktopDivId .previousnext-next-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
      test("it sets the mobile next url appropriately", () => {
        const expectedUrl = "/for-partners/lenders/7a-loans/operate-as-a-7a-lender";
        const component = shallow(<PreviousNext lineage={modifiedLineageForPartners} />);
        expect(component.find("#mobileDivId .previousnext-next-url").props().href).toBe(expectedUrl);
      });
    });
    describe("When in the last article of a section", () => {
      const modifiedLineageForPartners = _.merge(_.cloneDeep(lineageForPartners), [{},{},{},_.last(lineageForPartners[2].children)]);
      test("it renders properly", () => {
        const component = renderer.create(<PreviousNext lineage={modifiedLineageForPartners}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
      test("it does not have a Next Article element", () => {
        const component = shallow(<PreviousNext lineage={modifiedLineageForPartners} />);
        expect(component.find("#desktopDivId .previousnext-next-url").exists()).toBe(false);
      });
      test("it sets the desktop previous url appropriately", () => {
        const expectedUrl = "/for-partners/lenders/7a-loans/training-and-courses";
        const component = shallow(<PreviousNext lineage={modifiedLineageForPartners} />);
        expect(component.find("#desktopDivId .previousnext-previous-url SmallSecondaryButton").props().url).toBe(expectedUrl);
      });
    });
  });
});
