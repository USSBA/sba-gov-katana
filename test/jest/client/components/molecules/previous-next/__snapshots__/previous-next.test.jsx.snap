// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousNext Business Guide When in a middle article it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous false"
  >
    <h6>
      Previous
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <i
        aria-hidden="true"
        className="fa fa-chevron-left"
      />
      <span>
        Pick your business location
      </span>
    </a>
  </div>
  <div
    className="next false"
  >
    <h6>
      Next
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <span>
        Choose your business name
      </span>
      <i
        aria-hidden="true"
        className="fa fa-chevron-right"
      />
    </a>
  </div>
</div>
`;

exports[`PreviousNext Business Guide When in the first article of a middle section it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous false"
  >
    <h6>
      Previous
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <i
        aria-hidden="true"
        className="fa fa-chevron-left"
      />
      <span>
        Buy an existing business or franchise
      </span>
    </a>
  </div>
  <div
    className="next false"
  >
    <h6>
      Next
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <span>
        Choose a business structure
      </span>
      <i
        aria-hidden="true"
        className="fa fa-chevron-right"
      />
    </a>
  </div>
</div>
`;

exports[`PreviousNext Business Guide When in the first article of the first section it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous invisible"
  >
    <h6>
      Previous
    </h6>
  </div>
  <div
    className="next false"
  >
    <h6>
      Next
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <span>
        Write your business plan
      </span>
      <i
        aria-hidden="true"
        className="fa fa-chevron-right"
      />
    </a>
  </div>
</div>
`;

exports[`PreviousNext Business Guide When in the last article of a middle section it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous false"
  >
    <h6>
      Previous
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <i
        aria-hidden="true"
        className="fa fa-chevron-left"
      />
      <span>
        Open a business bank account
      </span>
    </a>
  </div>
  <div
    className="next false"
  >
    <h6>
      Next
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <span>
        Manage your finances
      </span>
      <i
        aria-hidden="true"
        className="fa fa-chevron-right"
      />
    </a>
  </div>
</div>
`;

exports[`PreviousNext Business Guide When in the last article of the last section it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous false"
  >
    <h6>
      Previous
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <i
        aria-hidden="true"
        className="fa fa-chevron-left"
      />
      <span>
        Native American-owned businesses
      </span>
    </a>
  </div>
  <div
    className="next invisible"
  >
    <h6>
      Next
    </h6>
  </div>
</div>
`;

exports[`PreviousNext For Partners When in a middle article it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous false"
  >
    <h6>
      Previous
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <i
        aria-hidden="true"
        className="fa fa-chevron-left"
      />
      <span>
        Operate as a 7(a) Lender
      </span>
    </a>
  </div>
  <div
    className="next false"
  >
    <h6>
      Next
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <span>
        Training and Courses
      </span>
      <i
        aria-hidden="true"
        className="fa fa-chevron-right"
      />
    </a>
  </div>
</div>
`;

exports[`PreviousNext For Partners When in the first article of a section it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous invisible"
  >
    <h6>
      Previous
    </h6>
  </div>
  <div
    className="next false"
  >
    <h6>
      Next
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <span>
        Operate as a 7(a) Lender
      </span>
      <i
        aria-hidden="true"
        className="fa fa-chevron-right"
      />
    </a>
  </div>
</div>
`;

exports[`PreviousNext For Partners When in the last article of a section it renders properly 1`] = `
<div
  className="previousNext"
  id="previous-next"
>
  <div
    className="previous false"
  >
    <h6>
      Previous
    </h6>
    <a
      className="button"
      onClick={[Function]}
      style={Object {}}
      target="_self"
    >
      <i
        aria-hidden="true"
        className="fa fa-chevron-left"
      />
      <span>
        Training and Courses
      </span>
    </a>
  </div>
  <div
    className="next invisible"
  >
    <h6>
      Next
    </h6>
  </div>
</div>
`;

exports[`PreviousNext Won't render without a lineage 1`] = `"received is not a function"`;
